addon/authenticators/oauth2-password-grant.js

```diff
@@ -434,17 +424,24 @@ export default BaseAuthenticator.extend({
     const serverTokenEndpoint = this.get('serverTokenEndpoint');
     return new RSVP.Promise((resolve, reject) => {
       this.makeRequest(serverTokenEndpoint, data).then((response) => {
-        run(() => {
-          expiresIn = response['expires_in'] || expiresIn;
-          refreshToken = response['refresh_token'] || refreshToken;
-          const expiresAt = this._absolutizeExpirationTime(expiresIn);
-          const data = assign(response, { 'expires_in': expiresIn, 'expires_at': expiresAt, 'refresh_token': refreshToken });
-          this._scheduleAccessTokenRefresh(expiresIn, null, refreshToken);
-          this.trigger('sessionDataUpdated', data);
-          resolve(data);
-        });
-      }, (response) => {
-        warn(`Access token could not be refreshed - server responded with ${response.responseJSON}.`, false, { id: 'ember-simple-auth.failedOAuth2TokenRefresh' });
+        deprecatePlainReturnValuesFromMakeRequest(response);
+
+        const rejectWithResponse = this.get('rejectWithResponse');
+        deprecateNotRejectingWithResponse(rejectWithResponse);
+
+        this._parseResponse(response, rejectWithResponse).then((json) => {
+          run(() => {
+            expiresIn = json['expires_in'] || expiresIn;
+            refreshToken = json['refresh_token'] || refreshToken;
+            const expiresAt = this._absolutizeExpirationTime(expiresIn);
+            const data = assign(json, { 'expires_in': expiresIn, 'expires_at': expiresAt, 'refresh_token': refreshToken });
+            this._scheduleAccessTokenRefresh(expiresIn, null, refreshToken);
+            this.trigger('sessionDataUpdated', data);
+            resolve(data);
+          }).catch(reject);
+        }).catch(reject);
+      }, (body) => {
+        warn(`Access token could not be refreshed - server responded with ${body}.`, false, { id: 'ember-simple-auth.failedOAuth2TokenRefresh' });
```

Comment:
I think the warning is only shown now when the fetch fails vs before that it was shown when the request failed, is that OK? (I don't know how important this warning is)